### DISCUSSION TITLE
speed parameter in urban

### DISCUSSION THREAD

**Discussion by ivantrouvetout** [Project-OSRM/osrm-backend#6845] [Q&A] [answered]
in the file, we find a speed association following the roads

     speeds = Sequence {
         -- road speeds
       speeds = Sequence {
       highway = {
         motorway = 90,--motorway
         motorway_link = 45,--ramp
         trunk = 85,--fast track
         primary = 65 --primary road
    }
This speed changes enormously depending on whether you are in town or in the countryside.
For example, a primary road in the countryside must be 80 km/h and in a dense city 40 km/h.

I found a contribution from "osrm-profiles-contrib" which takes this parameter into account.

[https://github.com/Project-OSRM/osrm-profiles-contrib/tree/master/5/18/urban-density](https://github.com/Project-OSRM/osrm-profiles-contrib/tree/master/5/18/urban-density)

Will this be more accurate than those offered by default ?

**Comment by DennisOSRM** [comment, marked as answer]
> Will this be more accurate than those offered by default ?

Generally speaking the answer is yes and no. It often depends on the situation on the ground and often _more_ specific rules apply to a region only. For example, in the area where I used to live there were primary road in a city you could easily (and legally) drive 80 kph on, while the same type of road would only allow for 50 nearby. Ideally, all roads would come with their true free flow speeds in the map data. But that's a science of its own.
